bbq/compiler/desugar.go

```diff
@@ -1028,22 +1047,22 @@ func (d *Desugar) interfaceDelegationMethodCall(
 }
 
 func (d *Desugar) addImport(location common.Location) {
-	interfaceLocation, isAddressLocation := location.(common.AddressLocation)
+	addressLocation, isAddressLocation := location.(common.AddressLocation)
```

Comment:
Unrelated to this PR: Is there a particular reason why we only handle address imports? Other imports seem to be ignored. Is that a TODO?

Reply:
Good question! I haven't run into to a scenario yet where the location is different. i.e: if the inherited codes are from a different program, I think it's always a deployed contract (given the locations are "resolved" by the time we reach here), and hence it's an address-location. But probably good to add some assertion here 👌

Reply:
Good question! I haven't run into to a scenario yet where the location is different. i.e: if the inherited codes are from a different program, I think it's always a deployed contract (given the locations are "resolved" by the time we reach here), and hence it's an address-location. But probably good to add some assertion here 👌

Reply:
Added in https://github.com/onflow/cadence/pull/3884/commits/0c2c4dea2264126516d461d3d4b1fc234a6bcafe